src/main/java/assignment2/services/DisplayDetails.java

```diff
@@ -0,0 +1,61 @@
+package assignment2.services;
+
+import assignment2.constants.Constants;
+import assignment2.exceptions.InputValidationException;
+import assignment2.exceptions.OrderMismatchException;
+import assignment2.models.User;
+import assignment2.utils.InteractiveScanning;
+import assignment2.utils.SortUsers;
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DisplayDetails {
+
+  final static Logger logger = LoggerFactory.getLogger(DisplayDetails.class);
+
+  /**
+   * Simply prints a line divider
+   */
+  private static void printLine() {
+    System.out.println("-----------------------------------------------------------------------");
+  }
+
+  /**
+   * Simply prints all the attributes of the user
+   */
+  private static void printHeaders() {
+    printLine();
+    System.out.println(
+        "Name" + Constants.TAB_SEPARATOR + "Roll number" + Constants.TAB_SEPARATOR + "Age"
+            + Constants.TAB_SEPARATOR + "Address" + Constants.TAB_SEPARATOR + "Courses");
+    printLine();
+  }
+
+  /**
+   * Displays the Users in the order chosen.
+   *
+   * @param users ArrayList of User objects
+   * @param scan  Scanner object to scan from cmd line
+   */
+  public static void display(ArrayList<User> users, Scanner scan) {
```

Comment:
method is to display what?

Reply:
This method is like a wrapper to the other methods which scan, print headers, etc. Then all of them are passed to the sort function which sorts based on user input and displays